### PR TITLE
Document `uniform_paths`

### PR DESCRIPTION
**Pull request by mattheww**
This updates use-declarations.md to describe the uniform_paths behaviour (requested in #487).

It also reorders things so that the main text and example describe the 2018 edition semantics, and the "Edition Differences" section says how the 2015 edition differs.

See also #291 "Path and module changes".

### CODE REVIEW

File: src/items/use-declarations.md [hunk 1]
```diff
@@ -82,31 +82,44 @@ In this example, the module `quux` re-exports two public names defined in
...
 
-An example of what will and will not work for `use` items:
-<!-- Note: This example works as-is in either 2015 or 2018. -->
+`crate` refers to the current crate. `self` refers to the current module.
+`super` refers to the parent module. `::` refers to the [extern prelude] (so
+it requires a crate name to follow).
 
-```rust
+If the path starts with an identifier, the identifier is resolved in the same
+way as for a path in an expression (in particular, it can be a crate name or
```
Comment (petrochenkov) [inline src/items/use-declarations.md]: "Expression" here is a bit too specific, this is true for any non-import path.

Comment (petrochenkov) [reply src/items/use-declarations.md]: >If the path starts with an identifier

And if the path doesn't start with an identifier, the behavior is also the same.
The text leaves the non-identifier case up in the air though and never talks about it below.

**Comment by mattheww** [comment]
I've pushed a new proposed version of the text.

I've reduced the text which describes resolution behaviour which isn't specific to `use` paths.

I've made it more explicit about which pieces of text describe which editions.

**Comment by mattheww** [comment]
Closing: #1052 covers this properly

**Pull request closed without merging**